### ISSUE TITLE
Unclickable area behind Plank

### ISSUE THREAD

**Issue by oredaze** [fvwmorg/fvwm3#612]
The issue is simple - when using the Plank dock there is an area behind it (even when the dock is hidden) that is unclickable and spans the whole screen. When I unmanage plank it is fixed, but then randomly the dock goes to the bottom layer, which makes it unusable.
I tried all of the EWMHM options and what not, but I conclude this has to be fixed in the code.
For a little bit more help - this issue is not present in openbox, but is present in other things like awesome. So I would understand if you say that it's not up to fvwm to fix this.

**Comment by ThomasAdam** [comment]
Hi @oredaze 

Hmm.  This is how plank looks for me:

![h](https://user-images.githubusercontent.com/101281/133304606-48b535f4-8f2f-47fd-98c9-6b8c075cc3af.png)

I'm not sure I'm seeing the same problem you are?

**Comment by oredaze** [comment]
It's not about how it looks, it's the functionality. I mean can you click the icons at the bottom-right? Actually the root menu of fvwm is a good way to test it. Click on the background progressively closer to the bottom of the screen and see when the menu stops activating.

**Comment by oredaze** [comment]
> Hi @oredaze
> 
> Hmm. This is how plank looks for me:
> 
> ![h](https://user-images.githubusercontent.com/101281/133304606-48b535f4-8f2f-47fd-98c9-6b8c075cc3af.png)
> 
> I'm not sure I'm seeing the same problem you are?

**Comment by ThomasAdam** [comment]
Weird, I cannot reproduce this -- I can click within mere pixels, and still I get a menu popup.

Can you fetch me the output of `xprop` on the Plank window, please?

**Comment by oredaze** [comment]
_NET_WM_STRUT(CARDINAL) = 0, 0, 0, 0
_NET_WM_STRUT_PARTIAL(CARDINAL) = 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
_NET_FRAME_EXTENTS(CARDINAL) = 0, 0, 0, 0
_KDE_NET_WM_FRAME_STRUT(CARDINAL) = 0, 0, 0, 0
_NET_WM_ALLOWED_ACTIONS(ATOM) = _NET_WM_ACTION_CHANGE_DESKTOP, _NET_WM_ACTION_CLOSE, _NET_WM_ACTION_MOVE, _NET_WM_ACTION_SHADE, _NET_WM_ACTION_STICK
_NET_WM_ICON_VISIBLE_NAME(UTF8_STRING) = "plank"
WM_STATE(WM_STATE):
		window state: Normal
		icon window: 0x0
_NET_WM_VISIBLE_NAME(UTF8_STRING) = "plank"
_NET_WM_DESKTOP(CARDINAL) = 4294967295
_NET_WM_STATE(ATOM) = _NET_WM_STATE_SKIP_PAGER, _NET_WM_STATE_SKIP_TASKBAR, _NET_WM_STATE_STICKY
WM_HINTS(WM_HINTS):
		Client accepts input or input focus: False
		Initial state is Normal State.
		window id # of group leader: 0x2000001
_GTK_THEME_VARIANT(UTF8_STRING) = 
XdndAware(ATOM) = BITMAP
_MOTIF_WM_HINTS(_MOTIF_WM_HINTS) = 0x2, 0x0, 0x0, 0x0, 0x0
_NET_WM_WINDOW_TYPE(ATOM) = _NET_WM_WINDOW_TYPE_DOCK
_NET_WM_SYNC_REQUEST_COUNTER(CARDINAL) = 33554437, 33554438
_NET_WM_USER_TIME_WINDOW(WINDOW): window id # 0x2000004
WM_CLIENT_LEADER(WINDOW): window id # 0x2000001
_NET_WM_PID(CARDINAL) = 10406
WM_LOCALE_NAME(STRING) = "en_US.UTF-8"
WM_CLIENT_MACHINE(STRING) = "ganymede"
WM_NORMAL_HINTS(WM_SIZE_HINTS):
		program specified minimum size: 63 by 900
		program specified maximum size: 63 by 900
		program specified base size: 0 by 0
		window gravity: NorthWest
WM_PROTOCOLS(ATOM): protocols  WM_DELETE_WINDOW, WM_TAKE_FOCUS, _NET_WM_PING, _NET_WM_SYNC_REQUEST
WM_CLASS(STRING) = "plank", "Plank"
WM_ICON_NAME(STRING) = "plank"
_NET_WM_ICON_NAME(UTF8_STRING) = "plank"
WM_NAME(STRING) = "plank"
_NET_WM_NAME(UTF8_STRING) = "plank

**Comment by oredaze** [comment]
NOTE:  _NET_WM_STATE_ABOVE failed to activate just now; I had to restart fvwm. I have `Layer 6` of course. Idk what was that.

**Comment by ThomasAdam** [comment]
@oredaze 

It's more effort on my part, but that's OK.

Can you send me through your fvwm config file(s).

**Comment by oredaze** [comment]
[config.gz](https://github.com/fvwmorg/fvwm3/files/7176091/config.gz)
IMPORTANT: I just realized it only happens if I have compositing on (picom in this case).

**Comment by ThomasAdam** [comment]
Oh right -- now this makes perfect sense. 

With a compositor enabled, this won't ever work due to how the root window clicks are interpreted.

**Comment by oredaze** [comment]
It's not just the root window, any window that sits behind plank is unclickable, but only if plank is above them in the stack.

**Comment by ThomasAdam** [comment]
Yes, sure.  Same difference, unfortunately.

**Comment by oredaze** [comment]
If there is nothing to be done I guess I am closing the issue. Thanks anyway.

**Comment by ThomasAdam** [comment]
Sorry.

**Comment by Antiz96** [comment]
Hi !

I've had the exact same issue with picom and plank on IceWM (unclickable area around plank window).
I've played a lot with the picom.conf file without success, but I've been able to solve this issue using the "winoption" option from IceWM (just found the solution a few minutes ago :dancers: ) !

Indeed, there's an option which allows you to change the default layer of a particular window (the "layer" option, see the man page for more info : https://ice-wm.org/man/icewm-winoptions).
So I just put the plank window in the "Below" layer by adding this line in my icewm's config file :
plank.layer: Below

Basically, you just need to put your plank window in a different layer than the top one used by your other normal windows.

I don't know if such option exists in fvwm, but as there's not a lot of topics about this particular issue and no given solution for now, I felt I had to give you mine :)

It also took me quite some time to figure this out... Hope this helps in any way !
